### PR TITLE
Use folly::MoveOnly in PolyTest.cpp

### PR DESCRIPTION
**Pull request by JoeLoser**
Summary:
- Use the `folly::MoveOnly` type in `PolyTest.cpp` instead of defining
  a separate type.

### CODE REVIEW

**Comment by yfeldblum** [comment]
When building with Clang,

```
folly/test/PolyTest.cpp:608:21: error: no viable conversion from 'folly::MoveOnly' (aka 'folly::moveonly_::MoveOnly') to 'Poly<folly::poly::IMoveOnly>'
    Poly<IMoveOnly> p = MoveOnly{};
                    ^   ~~~~~~~~~~
```

**Comment by JoeLoser** [comment]
Sigh, burned again by the fact that [this test isn't built](https://github.com/facebook/folly/blob/master/CMakeLists.txt#L822) by the open source build which is why I didn't catch the compiler error. I'll look into it. Thanks.

I'll create a PR in the near future to enable many missing tests from `folly/test` to be built/run by the open-source build to avoid these sort of things in the future.

**Comment by yfeldblum** [comment]
When building with Clang,

```
folly/test/PolyTest.cpp:609:35: error: temporary of type 'folly::MoveOnly' (aka 'folly::moveonly_::MoveOnly') has protected destructor
    Poly<IMoveOnly> p = MoveOnly_{};
```

**Comment by JoeLoser** [comment]
I couldn't recreate that compiler issue locally with Clang 7.0.1 or Clang 8.0.0. This [LLVM Code Review](https://reviews.llvm.org/D53860) may be the culprit.

I'm not as pleased with the change's benefit as the original code now, but it's a tiny better over the existing code. 

It did expose some linker errors on my end with this test which I have not yet chased down. Given that we'll have a Mac OS X CI soon, I added a new tag for tests which are disabled on Mac OS X, similar to the `WINDOWS_DISABLED`. I marked this test with the new tag.

File: folly/test/PolyTest.cpp [hunk 1]
```diff
@@ -614,6 +605,9 @@ TEST(Poly, Move) {
     EXPECT_EQ(&poly_cast<int>(p), &poly_cast<int>(q));
   }
   {
+    struct MoveOnly_ : MoveOnly {
+      virtual ~MoveOnly_() = default;
```
Comment (yfeldblum) [inline folly/test/PolyTest.cpp]: Does this need to be `virtual`?

Comment (JoeLoser) [reply folly/test/PolyTest.cpp]: No, sorry. Fixed.

**Pull request closed without merging**